apps/prairielearn/src/pages/instructorAssessments/instructorAssessments.html.tsx

```diff
@@ -196,3 +199,189 @@ export function AssessmentStats({ row }: { row: AssessmentStatsRow }) {
+                                output: row.sync_errors,
+                              })
+                            : row.sync_warnings
+                              ? SyncProblemButton({
+                                  type: 'warning',
+                                  output: row.sync_warnings,
+                                })
+                              : ''}
+                          <a href={`${urlPrefix}/assessment/${row.id}/`}>
+                            {row.title}
+                            {row.group_work && <i class="fas fa-users" aria-hidden="true"></i>}
+                          </a>
+                          <span
+                            dangerouslySetInnerHTML={{
+                              __html: IssueBadge({
+                                count: row.open_issue_count,
+                                urlPrefix,
+                              }).toString(),
+                            }}
+                          />
+                        </td>
+                        <td class="align-middle">{row.tid}</td>
+                        {PreactAssessmentStats({ row })}
+                      </tr>
+                    </Fragment>
+                  ))}
+                </tbody>
+              </table>
+            </div>
+
+            <div class="card-footer">
+              Download
+              <a href={`${urlPrefix}/instance_admin/assessments/file/${csvFilename}`}>
+                {csvFilename}
+              </a>
+              (includes more statistics columns than displayed above)
+            </div>
+          </div>
+        </main>
+      </body>
+    </html>,
+  );
+}
+
+function PreactAssessmentStats({ row }: { row: AssessmentStatsRow }) {
+  const spinner = (
+    <div class="spinner-border spinner-border-sm" role="status">
+      <span class="sr-only">Loading...</span>
+    </div>
+  );
+
+  return (
+    <Fragment>
+      <td class="text-center align-middle score-stat-number" style="white-space: nowrap;">
+        {row.needs_statistics_update ? spinner : row.score_stat_number}
+      </td>
+
+      <td class="text-center align-middle score-stat-score-hist" style="white-space: nowrap;">
+        {row.needs_statistics_update ? (
+          spinner
+        ) : row.score_stat_number > 0 ? (
+          <div
+            class="js-histmini d-inline-block"
+            data-data={JSON.stringify(row.score_stat_hist)}
+            data-options={JSON.stringify({ width: 60, height: 20 })}
+          />
+        ) : (
+          <Fragment>&mdash;</Fragment>
+        )}
+      </td>
+
+      <td class="text-center align-middle score-stat-mean" style="white-space: nowrap;">
+        {row.needs_statistics_update ? (
+          spinner
+        ) : row.score_stat_number > 0 ? (
+          <div class="d-inline-block align-middle" style="min-width: 8em; max-width: 20em;">
+            {/* TODO: This needs to be a Preact component */}
+            {Scorebar(Math.round(row.score_stat_mean))}
```

Comment:
How does this work? I thought we'd need to use the `dangerouslySetInnerHTML` trick to include a non-Preact component?

Reply:
This doesn't work 😂 this code path was just never hit locally because there weren't any submissions to show stats for. I skipped this for now since it wasn't necessary to get the render timing I was after.